test/arcade/patch/agent/cell/PatchCellCARTTest.java

```diff
@@ -0,0 +1,235 @@
+import org.junit.jupiter.api.Test;
+import sim.engine.SimState;
+import sim.util.Bag;
+import ec.util.MersenneTwisterFast;
+import arcade.core.agent.cell.CellState;
+import arcade.core.env.location.Location;
+import arcade.core.sim.Simulation;
+import arcade.core.util.MiniBox;
+import arcade.core.util.Parameters;
+import arcade.patch.env.grid.PatchGrid;
+import arcade.patch.env.location.PatchLocation;
+import arcade.patch.util.PatchEnums;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static arcade.core.ARCADETestUtilities.randomDoubleBetween;
+import static arcade.core.ARCADETestUtilities.randomIntBetween;
+
+public class PatchCellCARTTest {
+
+    private PatchCellCART patchCellCART;
+
+    private Parameters parameters;
+
+    private PatchLocation location;
+
+    private PatchCellContainer container;
+
+    static class PatchCellMock extends PatchCellCART {
+        PatchCellMock(PatchCellContainer container, Location location, Parameters parameters) {
+            super(container, location, parameters, null);
+        }
+
+        @Override
+        public PatchCellContainer make(int newID, CellState newState, MersenneTwisterFast random) {
+            return new PatchCellContainer(
+                    newID,
+                    id,
+                    pop,
+                    age,
+                    divisions,
+                    newState,
+                    volume,
+                    height,
+                    criticalVolume,
+                    criticalHeight);
+        }
+
+        @Override
+        public void step(SimState state) {}
+    }
+
+    @BeforeEach
+    public void setUp() throws NoSuchFieldException, IllegalAccessException {
+        parameters = spy(new Parameters(new MiniBox(), null, null));
+        location = mock(PatchLocation.class);
+
+        doReturn(0.0).when(parameters).getDouble(any(String.class));
+        doReturn(0).when(parameters).getInt(any(String.class));
+
+        int id = 1;
+        int parentId = 1;
+        int pop = 1;
+        int age = randomIntBetween(1, 120950);
+        int divisions = 10;
+        double volume = randomDoubleBetween(100, 200);
+        double height = randomDoubleBetween(4, 10);
+        double criticalVolume = randomDoubleBetween(100, 200);
+        double criticalHeight = randomDoubleBetween(4, 10);
+        PatchEnums.State state = PatchEnums.State.UNDEFINED;
+        ;
+
+        container =
+                new PatchCellContainer(
+                        id,
+                        parentId,
+                        pop,
+                        age,
+                        divisions,
+                        state,
+                        volume,
+                        height,
+                        criticalVolume,
+                        criticalHeight);
+
+        when(parameters.getDouble("HETEROGENEITY")).thenReturn(0.0);
```

Comment:
I think in your set up, you can give some default dummy value for all parameters, and then only change the relevant ones for the test. I think this will help you clean up your tests quite a bit.